connect/src/com/telenor/mobileconnect/SimCardStateChangedBroadcastReceiver.java

```diff
@@ -0,0 +1,35 @@
+package com.telenor.mobileconnect;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+
+import com.telenor.connect.ConnectSdk;
+
+public class SimCardStateChangedBroadcastReceiver extends BroadcastReceiver {
+
+    private static SimCardStateChangedBroadcastReceiver sSimCardChangedReceiver =
+            new SimCardStateChangedBroadcastReceiver();
+
+    private static IntentFilter SIM_CARD_CHANGED_INTENT_FILTER =
+            new IntentFilter("android.intent.action.SIM_STATE_CHANGED");
+
+    private SimCardStateChangedBroadcastReceiver() {
+    }
+
+    public static SimCardStateChangedBroadcastReceiver getReceiver() {
+        return sSimCardChangedReceiver;
+    }
+    public static IntentFilter getIntentFilter() {
+        return SIM_CARD_CHANGED_INTENT_FILTER;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String state = intent.getExtras().getString("ss");
+        if (state != null && "READY".equals(state)) {
+            ConnectSdk.sdkReinitializeMobileConnect();
```

Comment:
Not much, since this is just a flag. The current auth. flow will be allowed to continue with the old config. A new auth. flow will reinitialize itself.